src/atom-environment.coffee

```diff
@@ -229,6 +229,10 @@ class AtomEnvironment extends Model
 
     checkPortableHomeWritable()
 
+    # Patch the `process.env` on startup to fix the problem first documented
+    # in #4126
+    process.env = @project.getEnv()
```

Comment:
This is going to make it less clean, but this should happen prior to packages being loaded. I.e. prior to @project being defined. Think of all the activities that packages may do on activation that they _should not be doing_ (but still are doing)n that are dependent on `process.env`.